bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -1444,17 +1460,23 @@ void sendAddSuccessCallbacks() {
                 return;
             }
             // Check if it is the next entry in the sequence.
-            if (pendingAddOp.entryId != 0 && pendingAddOp.entryId != lastAddConfirmed + 1) {
+            if (pendingAddOp.entryId != 0 && pendingAddOp.entryId != pendingAddsSequenceHead) {
```

Comment:
I would defer introducing `pendingAddsSequenceHead` until `LastAddSynced` is introduced. At this moment, there is no `LastAddSynced`, so there is no reason to make change at this moment.

Reply:
*pendingAddsSequenceHead*  is the core of this  change. We have to handle the sequence of adds without advancing lastAddConfirmed

Reply:
sure. but my concern is no one is actually using this. I was thinking this pull request only for pushing the flag to the wire. to implementing the actual logic, you should do the server side first and client side later. Otherwise, you are pushing a lot of `unused` logic into the client, and we don't know when they are going to be enabled (given current pull requests are merged very quickly)

Reply:
as we have pushed server-side changes I will keep *pendingAddsSequenceHead*

Reply:
we decided to drop the concept of server side LastAddSynced, so this *pendingAddsSequenceHead* is fine in my opinion. IT is like what @jvrao is calling "nonDurableLAC"

Reply:
@sijie  Proposal to have nonDurableLAC is indeed to keep the LAC protocol unchanged. If we update LAC with non-durable LAC value, recovery becomes complicated because recovery assumes that up to LAC things are persisted.  As the piggy-backed LAC gets persisted on the disk, and if the client dies before closing the ledger, Current recovery open logic hunts for writes beyond LAC. But if LAC doesn't reflect persisted entry, then Recovery needs to go all the way to entry-id-0. If we update LAC only when persisted we keep the overall understanding of LAC intact.